ports/geckolib/gecko_bindings/sugar/refptr.rs

```diff
@@ -27,44 +37,62 @@ pub unsafe trait RefCounted where Self: Sized {
         };
 
         let result = cb(owned.as_ref());
+        // this isn't unwind safe, do not use without panic=abort
         forget(owned);
 
         result
     }
 
+    /// Given a non-null strong FFI reference, converts it into an Arc.
+    /// Panics on null.
     fn into(ptr: Strong<Self::FFIType>) -> Arc<Self> {
+        assert!(!ptr.is_null());
         unsafe { transmute(ptr) }
     }
 
+    /// Given a borrowed FFI reference, converts it to an Arc.
+    /// Unsafe to run on null values.
+    /// Can cause use-after-free if the Arc is allowed to have its destructor
+    /// run, please forget() arcs created by this
```

Comment:
I feel like we should avoid these "unsafe to run on null values" functions if at all possible, and return Option<T> instead. The callers can then unwrap them if they want to. Every safe use case for borrowed_into will need to check for null anyways. (A.K.A. I think this function should stay unsafe but respond with an Option<Arc<Self>> instead)

On a similar note, why not have the signature be:
```rust
fn borrowed_into(ptr: &Borrowed<Self::FFIType>) -> Option<&Arc<Self>>
```

As that way we can make it safe, and don't need to remind people to mem::forget() the Arc. (This does mean that release() will have to ptr::read() or similar out of the return value, but that is probably fine).